src/main/java/org/wise/portal/presentation/web/controllers/archive/ArchiveProjectController.java

```diff
@@ -0,0 +1,118 @@
+    UserTag userTag = userTagsService.get(user, ARCHIVED_TAG);
+    if (userTag == null) {
+      userTag = userTagsService.createTag(user, ARCHIVED_TAG);
+    }
+    userTagsService.applyTag(project, userTag);
+    return ResponseEntityGenerator.createSuccess(createProjectResponse(user, project));
+  }
+
+  @PutMapping("/projects/archived")
+  protected ResponseEntity<List<Map<String, Object>>> archiveProjects(
+      @RequestBody List<Long> projectIds) throws Exception {
+    User user = ControllerUtil.getSignedInUser();
+    UserTag userTag = userTagsService.get(user, ARCHIVED_TAG);
+    if (userTag == null) {
+      userTag = userTagsService.createTag(user, ARCHIVED_TAG);
+    }
+    List<Project> projects = getProjects(projectIds);
+    for (Project project : projects) {
+      userTagsService.applyTag(project, userTag);
+    }
+    return ResponseEntityGenerator.createSuccess(createProjectsResponse(user, projects));
+  }
+
+  @DeleteMapping("/project/{projectId}/archived")
+  protected ResponseEntity<Map<String, Object>> unarchiveProject(
+      @PathVariable("projectId") ProjectImpl project) throws JSONException {
+    User user = ControllerUtil.getSignedInUser();
+    UserTag userTag = userTagsService.get(user, ARCHIVED_TAG);
+    if (userTag != null) {
+      userTagsService.removeTag(project, userTag);
+    }
+    return ResponseEntityGenerator.createSuccess(createProjectResponse(user, project));
+  }
+
+  @DeleteMapping("/projects/archived")
+  protected ResponseEntity<List<Map<String, Object>>> unarchiveProjects(
+      @RequestParam List<Long> projectIds) throws Exception {
+    User user = ControllerUtil.getSignedInUser();
+    UserTag userTag = userTagsService.get(user, ARCHIVED_TAG);
+    List<Project> projects = getProjects(projectIds);
+    if (userTag != null) {
+      for (Project project : projects) {
+        userTagsService.removeTag(project, userTag);
+      }
+    }
+    return ResponseEntityGenerator.createSuccess(createProjectsResponse(user, projects));
+  }
+
+  private List<Project> getProjects(List<Long> projectIds) throws ObjectNotFoundException {
+    List<Project> projects = new ArrayList<Project>();
+    for (Long projectId : projectIds) {
+      projects.add(projectService.getById(projectId));
+    }
+    return projects;
+  }
+
+  private Map<String, Object> createProjectResponse(User user, Project project)
+      throws JSONException {
+    Map<String, Object> response = new HashMap<String, Object>();
+    response.put("id", project.getId());
+    response.put("archived", userTagsService.hasTag(user, project, ARCHIVED_TAG));
+    return response;
+  }
+
+  private List<Map<String, Object>> createProjectsResponse(User user, List<Project> projects)
+      throws JSONException {
+    List<Map<String, Object>> response = new ArrayList<Map<String, Object>>();
+    for (Project project : projects) {
+      response.add(createProjectResponse(user, project));
+    }
+    return response;
+  }
```

Comment:
Looks like these method do not actually throw a JSONException. Remove "throws" clause from these methods' prototypes and also from their callees in this class?